bitbots_body_behavior/bitbots_body_behavior/actions/head_modes.py

```diff
@@ -11,46 +15,38 @@ class AbstractHeadModeElement(AbstractActionElement):
 class LookAtBall(AbstractHeadModeElement):
     """Search for Ball and track it if found"""
     def perform(self):
-        self.blackboard.blackboard.set_head_duty(HeadMode.BALL_MODE)
-        return self.pop()
-
-
-class LookAtGoalPosts(AbstractHeadModeElement):
-    """Search for goal posts, mainly to locate the robot on the field"""
-    def perform(self):
-        self.blackboard.blackboard.set_head_duty(HeadMode.POST_MODE)
-        return self.pop()
-
-
-class LookAtBallAndGoalPosts(AbstractHeadModeElement):
-    """Track ball and goal by constantly switching between both"""
+        ball_position = self.blackboard.world_model.get_best_ball_position()
+        server_running = self.blackboard.lookat_action_client.wait_for_server(timeout_sec=1.0)
+        if not server_running:
+            while not server_running and rclpy.ok():
+                self.blackboard.node.get_logger().warn(
+                                      "Lookat Action Server not running! Lookat cannot work without lookat server!"
+                                      "Will now wait until server is accessible!",
+                                      throttle_duration_sec=10.0)
+                server_running = self.blackboard.lookat_action_client.wait_for_server(timeout_sec=1)
+            if server_running:
+                self.blackboard.node.get_logger().warn("Lookat server now running, 'look_at_ball' action will go on.")
+            else:
+                self.blackboard.node.get_logger().warn("Lookat server did not start.")
+
+        goal = LookAt.Goal()
+        goal.target_point = ball_position
+        self.blackboard.lookat_action_client.send_goal(goal) # TODO: when to use send_goal_async?
```

Comment:
Exactly. This dsd action pops itself after the the look at ros action has been completed, which is fine imo. You could add another parameter called `non-blocking` to the dsd action and call the async goal instead. This could be cool if we want to do other things during the head movement, but it is not important imo.

Reply:
Nvm. I would design it differently, comparable to the kick action, because the `perform` method in the dsd should not be blocked. We should use the async goal and check during every perform if the async task is finished. If it is, we pop ourselves. Here you can see how e.g. a done callback is registered: https://github.com/bit-bots/bitbots_behavior/blob/master/bitbots_blackboard/bitbots_blackboard/capsules/kick_capsule.py

Reply:
I changed it and the behavior does not crash in simulation. As we do not use this feature yet, I cannot test it further.